xgboost.py

```diff
@@ -0,0 +1,59 @@
+import sys
+sys.path.append("..")
+
+import pandas as pd
+import numpy as np
+from xgboost import XGBClassifier
+from sklearn.model_selection import StratifiedKFold,cross_val_score
+from sklearn.metrics import f1_score,classification_report,confusion_matrix
+
+from data_preprocessing.preprocessor import Preprocessor
+
+class FraudDetector:
+    def __init__(self,df):
+        self.df = df.copy()
+        self.X_nan = None
+        self.y_nan = None
+        self.df_dropped = None
+        self.X,self.y = None,None
+        self.X_train,self.X_test,self.y_train,self.y_test = None,None,None,None
+        self.xgb = None
+        self.skf = StratifiedKFold(n_splits=5,random_state=42,shuffle=True)
+        
+    def preprocess(self):
+        preprocessor = Preprocessor(self.df)
+        preprocessor.clean()
+        
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `preprocess` method creates a new `Preprocessor` object and calls its `clean` method, but it doesn't seem to do anything with the cleaned data. The cleaned data should be assigned back to `self.df` if the cleaning operation is done in-place.

```diff
-        preprocessor.clean()
+        self.df = preprocessor.clean()
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->